pipeline/test_etl.py

```diff
@@ -0,0 +1,232 @@
+
+    @patch("transform.requests.get")
+    def test_transform_get_genres_from_url(self, mock_get):
+        """Test genre extraction from a URL."""
+        mock_response = MagicMock(status_code=200, text='<a class="tag">Rock</a><a class="tag">Pop</a>')
+        mock_get.return_value = mock_response
+        locations = ["us", "rockville"]
+
+        result = get_genres_from_url("//example.com", locations)
+        assert set(result) == {"Rock", "Pop"}
+
+        mock_get.side_effect = Exception("Network Error")
+        assert get_genres_from_url("//example.com", locations) == []
+
+
+    @patch("transform.requests.get")
+    def test_transform_get_release_date_from_url(self, mock_get):
+        """Test release date extraction from a URL."""
+        mock_response = MagicMock(status_code=200, text='<meta name="description" content="released 1 October 2021">')
+        mock_get.return_value = mock_response
+        assert get_release_date_from_url("//example.com") == "2021-10-01"
+        mock_response.text = '<meta name="description" content="No release date">'
+        mock_get.return_value = mock_response
+        assert get_release_date_from_url("//example.com") == ""
+        mock_get.side_effect = Exception("Network Error")
+        assert get_release_date_from_url("//example.com") == ""
+
+    def test_transform_create_sales_dataframe_album(self):
+        """Test DataFrame creation for album sales."""
+        mock_sales_info = [
+            {
+                "release_type": "a",
+                "release_description": "Sample Album",
+                "album_title": "Mock Album",
+                "artist_name": "Mock Artist",
+                "country": "US",
+                "amount_paid_usd": 10.0,
+                "genres": ["Rock", "Pop"],
+                "release_date": "01-10-2021",
+                "sale_date": "2021-10-01",
+            }
+        ]
+        df = create_sales_dataframe(mock_sales_info)
+        assert isinstance(df, pd.DataFrame)
+        assert len(df) == 1
+        assert "release_type" in df.columns
+        assert df.iloc[0]["release_type"] == "a"
+
+    def test_transform_create_sales_dataframe_album(self):
+        """Test DataFrame creation for album sales."""
+        mock_sales_info = [
+            {
+                "release_type": "t",
+                "release_description": "Sample Track",
+                "album_title": "Mock Track",
+                "artist_name": "Mock Artist",
+                "country": "US",
+                "amount_paid_usd": 10.0,
+                "genres": ["Rock", "Pop"],
+                "release_date": "01-10-2021",
+                "sale_date": "2021-10-01",
+            }
+        ]
+        df = create_sales_dataframe(mock_sales_info)
+        assert isinstance(df, pd.DataFrame)
+        assert len(df) == 1
+        assert "release_type" in df.columns
+        assert df.iloc[0]["release_type"] == "t"
+
+
```

Comment:
Could we add a test to check abbreviations 'a' and 't' are replaced by 'album' and 'track'